md/minimizer.py

```diff
@@ -103,21 +108,36 @@ def minimize_host_4d(mols, host_system, host_coords, ff, box):
         u_impls.append(fn)
 
     # context components: positions, velocities, box, integrator, energy fxns
+
+    def min_func(x0, lamb):
+        x0 = x0.reshape((-1, 3))
+        u = 0
+        du_dx_sum = np.zeros_like(x0)
+        for impl in u_impls:
+            du_dx, _, u_bond = impl.execute(x0, box, lamb)
+            u += u_bond
+            du_dx_sum += du_dx
+        return u, du_dx_sum.reshape(-1)
+
+    options = {"disp": verbose, "maxiter": 25}
```

Comment:
25 iterations based on the CDK8 set I was running into issues with before. Most of the energy reduction took place in the first 20 iterations and if it ran for too long it would end up exploding. Could probably avoid the latter by adding constraints

Reply:
IIRC when we debug this together we mainly saw line searching blow up.

Reply:
Correct. My sense is that without constraints it is creating a 'bad' state that makes the gradients/energies no longer meaningful. The line searching would fail after big jumps in energy